find_posts.py

```diff
@@ -1025,6 +1026,132 @@ def __len__(self):
     def toJSON(self):
         return json.dump(self._dict, f, default=str)
 
+def get_server_from_host_meta(server):
+    url = f'https://{server}/.well-known/host-meta'
+    try:
+        resp = get(url, timeout = 30)
+    except Exception as ex:
+        log(f"Error getting host meta for {server}. Exception: {ex}")
+        return None
+
+    if resp.status_code == 200:
+        try:
+            hostMeta = ET.fromstring(resp.text)
+            lrdd = hostMeta.find(".//{http://docs.oasis-open.org/ns/xri/xrd-1.0}Link[@rel='lrdd']")
+            url = lrdd.get('template')
+            match = re.match(
+                r"https://(?P<server>[^/]+)/", url
+            )
+            if match is not None:
+                return match.group("server")
+            else:
+                raise Exception('server not found in lrdd')
+                return None
+        except Exception as ex:
+            log(f'Error parsing host meta for {server}. Exception: {ex}')
+            return None
+    else:
+        log(f'Error getting host meta for {server}. Status Code: {resp.status_code}')
+        return None
+
+def get_nodeinfo(server, host_meta_fallback = False):
+    url = f'https://{server}/.well-known/nodeinfo'
+    try:
+        resp = get(url, timeout = 30)
+    except Exception as ex:
+        log(f"Error getting host node info for {server}. Exception: {ex}")
+        return None
+
+    # if well-known nodeinfo isn't found, try to check host-meta for a webfinger URL
+    # needed on servers where the display domain is different than the web domain
```

Comment:
If `/.well-known/nodeinfo` doesn't reveal to us the web domain of the instance to fetch `/nodeinfo/2.0` from, is there a scenario where `/.well-known/host-meta` would? Otherwise AFAIK these should be identically configured and we can give up here on a failure.

Reply:
To demonstrate, here's the response for `/.well-known/nodeinfo/` on the display domain `shatteredsky.net`, and the response which includes the web domain `mastodon.shatteredsky.net`.

![image](https://github.com/nanos/FediFetcher/assets/5181964/3d2072bb-51bf-4c8c-a5b4-0c230b1a40ea)

Reply:
I'm honestly not sure. Though I do suspect that there'll be some software that will do this, so might be good to leave it in.

Reply:
Ok, makes sense. So the current implementation is the way forward, isn’t it?